### PR TITLE
Add code coverage using Istanbul and Coveralls

### LINKED ISSUE #200: Add code coverage
I like [Istanbul](https://github.com/gotwarlost/istanbul) but any other modern coverage tool is great too.

### PR DESCRIPTION
**Pull request by pgilad**
Fixes #200

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,14 +70,16 @@
...
     "start": "run-browser test/index.js",
-    "lint": "jshint --exclude benchmark/ .",
+    "lint": "jshint ampersand-state.js ./test/*",
     "benchmark": "node --allow-natives-syntax benchmark/massCreate.js",
     "preversion": "git checkout master && git pull && npm ls",
     "publish-patch": "npm run preversion && npm version patch && git push origin master --tags && npm publish",
     "publish-minor": "npm run preversion && npm version minor && git push origin master --tags && npm publish",
-    "publish-major": "npm run preversion && npm version major && git push origin master --tags && npm publish"
+    "publish-major": "npm run preversion && npm version major && git push origin master --tags && npm publish",
+    "coveralls": "cat ./coverage/lcov.info | coveralls"
```
Comment (wraithgar) [inline package.json]: Where is this file supposed to come from?

Comment (pgilad) [inline package.json]: It's generated by Istanbul

Comment (pgilad) [inline package.json]: Try running `npm run coverage`

**Comment by cdaringe** [comment]
happy to be adding coverage!  doesn't change any functional code, so if it works for you, +1 from me!

File: package.json (hunk 1 above)
Comment (wraithgar) [inline package.json]: Ah great! should we then make `npm run coveralls` first run `npm run coverage`?

Comment (pgilad) [inline package.json]: It's not considered best practice to couple npm scripts. It prevents correct orchestration... In `.travis.yml` I added the `npm run coveralls` as after_scripts as well

**Comment by wraithgar** [comment]
Thanks for doing this :shipit:

**Pull request merged**